src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php

```diff
@@ -221,133 +215,20 @@ public function addPlatformToTopology(PlatformTopology $platformTopology): void
-                    $errorMessage = sprintf(
-                        _("The platform: '%s'@'%s' cannot be added to the Central linked to this Remote"),
-                        $platformTopology->getName(),
-                        $platformTopology->getAddress()
-                    );
-                    $returnedMessage = json_decode($registerResponse->getContent(false), true);
-
-                    if (!empty($returnedMessage)) {
-                        $errorMessage .= "  /  " . _("Central's response => Code : ") .
-                            implode(', ', $returnedMessage);
-                    }
-                    throw new PlatformTopologyConflictException(
-                        $errorMessage
-                    );
-                }
-            } catch (TransportExceptionInterface $e) {
-                throw new PlatformTopologyException(
-                    _("Request to the Central's API failed") . (' : ') . $e->getMessage()
-                );
-            } catch (ClientExceptionInterface $e) {
-                throw new PlatformTopologyException(
-                    _("API calling the Central returned a Client exception") . (' : ') . $e->getMessage()
-                );
-            } catch (RedirectionExceptionInterface $e) {
-                throw new PlatformTopologyException(
-                    _("API calling the Central returned a Redirection exception") . (' : ') . $e->getMessage()
-                );
-            } catch (ServerExceptionInterface $e) {
-                $message = _("API calling the Central returned a Server exception");
-                if (!empty($optionPayload['proxy'])) {
-                    $message .= '. ' . _("Please check the 'Centreon UI' form and your proxy configuration");
-                }
-                throw new PlatformTopologyException(
-                    $message . (' : ') . $e->getMessage()
-                );
-            } catch (DecodingExceptionInterface $e) {
-                throw new PlatformTopologyException(
-                    _("Unable to convert Central's API response") . (' : ') . $e->getMessage()
-                );
-            } catch (\Exception $e) {
-                throw new PlatformTopologyException(
-                    _("Error from Central's register API") . (' : ') . $e->getMessage()
+                $this->platformTopologyRegisterRepository->registerPlatformTopologyToParent(
+                    $platformTopology,
+                    $foundPlatformInformation,
+                    $this->proxyService->getProxy()
                 );
+            } catch (PlatformTopologyConflictException $ex) {
+                throw $ex;
+            } catch (RepositoryException $ex) {
+                throw new PlatformTopologyException($ex->getMessage(), $ex->getCode(), $ex);
+            } catch (\Exception $ex) {
+                throw new PlatformTopologyException('', 0, $ex);
```

Comment:
not sure it is a good practice to let exception message empty (cc @callapa)

Reply:
indeed I forgot this one

Reply:
done
conflicts resolved and branch rebased